src/datasource.ts

```diff
@@ -8,34 +8,138 @@ import {
   TimeRange,
 } from '@grafana/data';
 import { getBackendSrv, getTemplateSrv } from '@grafana/runtime';
-
-import { lastValueFrom } from 'rxjs';
-
+import { lastValueFrom , Observable , Subject , of , from , map } from 'rxjs';
 import { calculateInterval, createEpochTimeRangeBoundary } from './utils/time.utils';
 import { SumoApiDimensionValuesResponse } from './types/metadataCatalogApi.types';
 import { mapKeyedErrorMessage } from './utils/keyedErrors.utils';
 import { MetricsQuery, SumoApiMetricsResponse, SumoQuery } from './types/metricsApi.types';
 import { dimensionsToLabelSuffix } from './utils/labels.utils';
 import { interpolateVariable } from './utils/interpolation.utils';
 
+import { formatSumoValues, getSumoToGrafanaType, SumoQueryType } from './types/constants';
+import { searchQueryExecutionService } from './services/logDataService/searchQueryExecutionService';
+import { RunSearchActionType, SearchQueryType, SearchStatus } from './services/logDataService/constants';
+import { IField, ISearchStatus, RunSearchAction, SearchQueryParams } from './services/logDataService/types';
+
+import { filter , switchMap } from 'rxjs/operators';
+
 export class DataSource extends DataSourceApi<SumoQuery> {
   private readonly baseUrl: string;
   private readonly basicAuth: string;
 
+  private logsController$ :  Subject<RunSearchAction> | null
+
   constructor(instanceSettings: DataSourceInstanceSettings) {
     super(instanceSettings);
     this.baseUrl = instanceSettings.url as string;
     this.basicAuth = instanceSettings.basicAuth as string;
+    this.logsController$ = null
   }
 
-  // perform metrics query by Grafana
-  async query(options: DataQueryRequest<SumoQuery>): Promise<DataQueryResponse> {
-    const { range, maxDataPoints, targets, scopedVars, interval } = options;
-    const templateSrv = getTemplateSrv();
+  // perform logs/metrics query by Grafana
+  query(options: DataQueryRequest<SumoQuery>): Observable<DataQueryResponse> {
+    const { range , targets } = options;
 
     if (!range) {
       throw new Error('No range has been provided');
     }
+    if(this.logsController$){
+      // stop previous running logs query
+      this.logsController$.next({ type : RunSearchActionType.STOP })
+      this.logsController$ = null;
+    }
+
+    //Check if any logs query present
+    const isLogsQuery = targets.find(query=>query.type === SumoQueryType.Logs)
+
+    if(isLogsQuery){
+      return this.fetchLogsQuery(options);
+    }
+    else{
+      return this.fetchMetricsQuery(options);
+    }
+  }
+
+
+  fetchLogsQuery(options: DataQueryRequest<SumoQuery>) : Observable<DataQueryResponse>{
```

Comment:
it looks like logs query does not support variables, and I think it should, in order to be consistent